File: satory074/abc157/b/main.cpp
#include <cstdio>
using namespace std;
int main() {
    // yoko
	int yoko[3][3];
    for (int i = 0; i < 3; i++){
        for (int j = 0; j < 3; j++){
            scanf("%d", &yoko[i][j]);
        }
    }

    // tate
    int tate[3][3];
    for (int i = 0; i < 3; i++){
        for (int j = 0; j < 3; j++){
            tate[i][j] = yoko[j][i]
        }
    }

    // naname
    int naname[2][3];
    for (int i = 0; i < 2; i++){
        for (int j = 0; j < 3; j++){
            naname[i][j] = yoko[j][j]
        }
    }


    int line[8][3];

    int
}

File: satory074/abc157/b/main.py
#!/usr/bin/env pypy3
import collections
import itertools as it
import math
import numpy as np
 
#  = input()
#  = int(input())
yoko = [list(map(int, input().split())) for _ in range(3)]
is_flag = [False] * 9
#  = list(map(int, input().split()))
#  = [int(input()) for i in range(N)]
#
# c = collections.Counter()

yoko = list(np.array(yoko).flatten())

n = int(input())
for _ in range(n):
    b = int(input())

    if b in yoko:
        is_flag[yoko.index(b)] = True

is_flag = list(np.array(is_flag).reshape(3, 3))

line = is_flag # yoko
line += list(np.array(is_flag).T) # tate
line += [[line[0][0], line[1][1], line[2][2]], [line[0][2], line[1][1], line[2][0]]]

for l in line:
    if sum(l) == 3:
        print('Yes')
        break
else:
    print('No')
#print(yoko)
#print(tate)
#print(naname)
